File: 2021/day01/solutions/arxcis.kt

fun main() {
  // Parse input
  val input = ArrayList<Int>()
  var rawLine = readLine()
  while (rawLine != null) {
    val ints = rawLine.split("\n").map{ it.toInt() }
    input.addAll(ints)
    rawLine = readLine()
  }

  // Part 1
  var count = 0
  for (left in 0 until input.size-1) {
    val right = left + 1
    count = if(input[left] < input[right]) ++count else count
  }
  println(count)

  // Part 2
  val measurements = ArrayList<Int>()
  for (one in 0 until input.size-2) {
    val two = one + 1
    val three = two + 1
    val sum = arrayOf(input[one],input[two],input[three]).sum()
    measurements.add(sum)
  }
  count = 0
  for (left in 0 until measurements.size-1) {
    val right = left + 1

    count = if(measurements[left] < measurements[right]) ++count else count
  }
  println(count)
}

File: 2021/day01/solutions/arxcis.rs
use std::{io, io::prelude::*};

fn main() -> io::Result<()> {
  let mut input = Vec::new();

  for line in io::stdin().lock().lines() {
    let num = line.unwrap().parse::<i32>().unwrap();
    input.push(num)
  }

  // Part 1
  let mut right_is_larger_count: i32 = 0;

  for i in 0..input.len() - 1 {
    let left = input[i];
    let right = input[i+1];

    right_is_larger_count += (left < right) as i32;
  }
  println!("{}", right_is_larger_count);

  // Part 2
  right_is_larger_count = 0;
  for i in 0..input.len() - 3 {
    let one = input[i];
    let two = input[i+1];
    let three = input[i+2];
    let four = input[i+3];

    let left = one + two + three;
    let right = two + three + four;

    right_is_larger_count += (left < right) as i32;
  }
  println!("{}", right_is_larger_count);

  Ok(())
}
